play-ebean/src/main/java/play/db/ebean/EbeanDynamicEvolutions.java

```diff
@@ -34,21 +33,27 @@ public class EbeanDynamicEvolutions extends DynamicEvolutions {
 
   private final Map<String, Database> databases = new HashMap<>();
 
-  @Inject
+  /**
+   * @deprecated No need to pass lifecycle anymore as shutdown is now managed by {@link
+   *     play.db.ebean.EbeanLifecycle}. Use {@link #EbeanDynamicEvolutions(EbeanConfig, Environment,
+   *     EvolutionsConfig)} instead.
+   */
+  @Deprecated
   public EbeanDynamicEvolutions(
       EbeanConfig config,
       Environment environment,
       ApplicationLifecycle lifecycle,
       EvolutionsConfig evolutionsConfig) {
+    this(config, environment, evolutionsConfig);
+  }
+
+  @Inject
+  public EbeanDynamicEvolutions(
+      EbeanConfig config, Environment environment, EvolutionsConfig evolutionsConfig) {
     this.config = config;
     this.environment = environment;
     this.evolutionsConfig = evolutionsConfig;
     start();
-    lifecycle.addStopHook(
-        () -> {
-          databases.forEach((key, database) -> database.shutdown(false, false));
-          return CompletableFuture.completedFuture(null);
-        });
```

Comment:
Why do you actually not just replace this code with the code you added in `EbeanLifecycle`? Why do you even need the `EbeanLifecycle` class? Isn't it much more straigt forward to change this code here and thats it?